### PR TITLE
[WIP]RAINCATCH-1156 Adding role based check to login

### PR DESCRIPTION
**Pull request by witmicko**
## Motivation
https://issues.jboss.org/browse/RAINCATCH-1156

## Description
Added allowed roles to the server config and cross check them with user roles (intersection of both size > 0) 

## Progress
- [x] Passport
- [ ] Unit tests
- [ ] Documentation

### CODE REVIEW

File: demo/server/config-dev.js [hunk 1]
```diff
@@ -12,14 +12,16 @@ var config = {
   "security": {
     "adminRole": "admin",
     "userRole": "user",
+    "allowedRoles": ["adminRole"],
```
Comment (wtrocki) [inline demo/server/config-dev.js:15]: -1. Roles can be specified directly in the places we need them. 
Why we need this in the server?

Comment (witmicko) [reply demo/server/config-dev.js:15]: would you agree to rename this to say `portalAllowedRoles`? I think we should avoid adding this to the client side.

**Comment by wtrocki** [comment]
@witmicko - We cannot have it this way as we provide role based support on top of the client application. We will need to move from admin and user roles fairly soon.

This means that users will need to use role based check - using our AuthService in the portal client to show dialog if user do not have permission as was suggested in the ticket. 
This change needs to work with both keycloak and passworrd. 
Using using interface will allow us to do so.

This aproach works but it's not using parts of our framework which we want others to use. 

ping @JameelB

**Comment by witmicko** [comment]
> This means that users will need to use role based check - using our AuthService in the portal client to show dialog if user do not have permission as was suggested in the ticket.

It does exactly, that. Not sure what is the problem here, pull changes locally and try this.

from what I saw so far passport and keycloak differ a lot in implementation.
We cant have role restriction on the client side - what is stopping me from changing it locally or modifying requests in preflight?

**Comment by wtrocki** [comment]
> from what I saw so far passport and keycloak differ a lot in implementation.

They using single interface that it should be used. If that's not possible then that is a bug as PRD clearly states that this should be possible.

**Comment by witmicko** [comment]
Closing for now in favour of client-side filtering (server side is still protected)

**Pull request closed without merging**